core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1020,16 +1024,31 @@ public void listStatus(AlluxioURI path, ListStatusContext context,
       ResultStream<FileInfo> resultStream)
       throws AccessControlException, FileDoesNotExistException, InvalidPathException, IOException {
     Metrics.GET_FILE_INFO_OPS.inc();
+    List<String> prefixComponents;
+    if (!context.getOptions().getPrefix().equals("")) {
+      // compute the prefix as path components, removing the first empty string
+      prefixComponents = Arrays.stream(PathUtils.getPathComponents(
```

Comment:
My 1st impression is you can use `PathUtils.hasPrefix(String path, String prefix)`. There is also a PrefixPathMatcher and a TrieNode class in alluxio code base, which i recently came to know. It's not super obvious, but I think we should reuse that. I'm also thinking about https://github.com/Alluxio/alluxio/issues/15266 If you will need a trie-flavor structure pls comment in that issue too so we gather more requirements.

Reply:
I can make a separate pull request for the CloseableIterator change.

I've moved most the prefix checks to use the PathUtils methods.

I'm happy to use a trie, I'm just not sure where it would be? Would this be like a separate structure in memory that represents the full file system so you could quickly check a path before locking the Inodes?

Reply:
Here is a PR that only changes the Iterable to CloseableIterator for ReadOnlyInodeStore. https://github.com/Alluxio/alluxio/pull/15455